common/Util-desktop.cpp

```diff
@@ -160,6 +162,43 @@ std::size_t getTotalSystemMemoryKb()
     return totalMemKb;
 }
 
+size_t getMaxConcurrentTCPConnections(const size_t cutOffMinimum)
+{
+#ifdef __linux__
+    auto readDecimal = [](const char* path, const size_t defaultValue) -> size_t {
+        FILE* file = fopen(path, "r");
+        if (file == nullptr)
+            return defaultValue;
+        char line[4096] = { 0 };
+        // coverity[tainted_data_argument : FALSE] - we trust the kernel-provided data
+        const char *line2 = fgets(line, sizeof(line), file);
+        fclose(file);
+        if(line2 == nullptr)
+            return defaultValue;
+        char* endptr = nullptr;
+        errno = 0;  // Flush previous error indicator. Reminder: errno is thread-local
+        const unsigned long long num = std::strtoull(line, &endptr, 10);
+        if (0 != errno || nullptr == endptr || endptr == line || num > std::numeric_limits<size_t>::max())
```

Comment:
0 != errno to errno  != 0 - as previously discussed. We use !ptr - ~everywhere for null pointer checking & readability - as above - please use that idiom - and above twice.

Reply:
Either way - I don't believe any of this error detection is helpful; if you read systemd you'll see they use sscanf() to parse /proc output - the format and it content is well defined and must be backwards compatible. Lets use stroll as is done in the functions above; and lets split this out into its own readDecimal function instead of having this in-line function - but lets give it a name that doesn't suck - and fits into the existing naming convention - I believe I asked for that before.

Reply:
Reanimating the `Util::getFromFile` you introduced last year .. adding a default value.